### DISCUSSION TITLE
PagerSlim for Lucene Search

### DISCUSSION THREAD

**Discussion by rjpowers10** [OrchardCMS/OrchardCore#6895] [Ideas]
I'm curious why the `Search` method in `OrchardCore.Lucene.Controllers.SearchController` returns a `PagerSlim` instead of a `Pager`. Is the performance really bad with a `Pager`? Are there any plans to use a `Pager` in the future? I think most users would expect to see a full pager for search results.

Sorry to question your design decisions, just figured I would ask :)

**Comment by jtkech** [comment]
Yes it is a lightweight pager, you may have to implement a custom search controller / view OR just override the `Search.cshtml` **mvc view** in your theme, as i remember under `.../Views/{module}/{controller}/{action}.cshtml`, so here `YourTheme/Views/OrchardCore.Lucene/Search/Search.cshtml`, and that would use a full pager.

With a lower precedence can also be just under `YourTheme/Views/Shared/Search.cshtml`, this is what is done under the blog theme but with a `Search.liquid` liquid file. Can also be defined in a module `YourModule/Views/Shared/Search.cshtml` if not defined in the theme that has precedence.

You can also override the `Search-Form.cshtml` and `Search-Results.cshtml` **shape view templates** under `YourTheme/Views` OR `YourModule/Views` if not defined in the current theme that has precedence.

**Comment by Skrypt** [comment]
PagerSlim has been used on the Search result page mainly because the full pager doesn't scale really well.
By default the PagerSlim will work but a full pager would require more work if we'd want it to be usable without any glitches in a scaled environment (multiple running docker containers for example).

Nothing prevents you though from using the full Pager by replacing the current one using the methods described by @jtkech in the previous comment.

**Reply by rjpowers10** [reply]
I agree that we could implement a custom search controller. I'm confused by the other answer which is to override Search.csthml. I understand how overriding views works but aren't I missing the information that I need to turn it into a full pager? The built in search controller returns a PagerSlim model so I don't think I have everything. Or am I not understanding the answer?

Either way, it's not critical for us. Thanks for taking the time to reply.

**Reply by Skrypt** [reply]
Yes, we assume you are working with source code and making the changes in the controller.

> you may have to implement a custom search controller

Or alternatively, eventually, the controller could use a theme / tenant setting which would make return a Pager or PagerSlim. If you want to contribute 😉
